### PR TITLE
Linux icon and pacman builds

### PR DESCRIPTION
**Pull request by jjgalvez**
I modified the app.js code to add an icon for Linux which was not appearing properly in gnome. I also added pacman to your list of builds which will allow this to be installed in Arch Linux based distros.

Thanks for writing this, hope this helps. I've only tested this in Arch .

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -16,6 +18,10 @@ const state = {
 const app = remote.app;
 const appDir = jetpack.cwd(app.getAppPath());
 
+if (IS_LINUX) {
+  app.mainWindow.setIcon(path.join(__dirname, '..', 'resources', 'icons', '128x128.png'))
+}
```
Comment (ajivoin) [inline src/app.js]: Probably worth checking if this has any unwanted effects on other Linux distros.  
I don't think I'll have the chance to check personally anytime soon, but if I do I'll add a comment here.

**Comment by ajivoin** [comment]
Here's a link to #33, the issue this is related to.

File: src/app.js (hunk 1 above)
Comment (chrisknepper) [reply src/app.js]: Definitely, though I'm surprised it didn't work out of the box on Arch because it seems to work on Ubuntu with GNOME.

**Comment by chrisknepper** [comment]
Thanks for this! I just installed Arch for the first time yesterday so I should be able to test this. Are you using GNOME or another DE?

**Comment by jjgalvez** [comment]
I'm using gnome, but I do have xfc installed as a secondary DE and could test it there if its helpful

**Comment by chrisknepper** [comment]
Can you post a screenshot of what you mean by that? On Ubuntu 18.04 with GNOME, the tray icon looks a little blurry on my display but otherwise appears correct.

**Comment by jjgalvez** [comment]
that's interesting the try icon does not work for me on Arch and I am using the appindicator gnome extension which I think is what Ubuntu uses.

**Comment by jjgalvez** [comment]
that's interesting the try icon does not work for me on Arch and I am using the appindicator gnome extension which I think is what Ubuntu uses. 

Here is a screenshot in XFC the tray icon looks doubled
tray

**Comment by jjgalvez** [comment]
here it is with gnome on arch, no tray icon. Again I have the appindicator extension installed and you can see that other apps have icons in the tray
no-tray

**Comment by chrisknepper** [comment]
I'm super-excited to get this working for Arch since there seems to be a contingent of Arch users! I like your code changes, but there must be a way to get the icons to show up on Linux without calling a function. According to [this comment](https://github.com/electron-userland/electron-builder/issues/2577#issuecomment-377562602), the presence of files named `icon.png` can interfere with creating Linux icons from the mac `.icns` file.

Looking at the [package.json of pb-for-desktop](https://github.com/sidneys/pb-for-desktop/blob/master/package.json), another popular Electron app, the icon set for Linux can be explicitly set to a folder.

Could you test this by removing the call to `setIcon` and adding this under the `linux` key in `package.json`:

`"icon": "./resources/icons"` ?

**Comment by jjgalvez** [comment]
just tried this and and it did not work for me, setting the icon in package.json did not set the icon correctly in linux

**Comment by chrisknepper** [comment]
@jjgalvez Seems to be a bug in electron-builder and your solution is what works. 

https://github.com/electron-userland/electron-builder/issues/2269#issuecomment-342168989

I have one more request and then we can merge. Could you move the icon setting code from `app.js` to `background.js`. It's a safer way to set the icon since it will be applied at the time of the mainWindow creation.

On line 86 of `background.js`, move the object from being defined inline in `createWindow` to a variable like:

```
const mainWindowOptions = {
    width: 1100,
    height: 800,
    autoHideMenuBar: settings.get('autoHideMenuPref')
};
```

Then

```
if (IS_LINUX) {
    mainWindowOptions.icon = path.join...
}
```

Then pass that to the call to `createWindow`.

I've tested in Ubuntu with GNOME with this and it works, about to test with Arch with GNOME.

**Comment by chrisknepper** [comment]
BTW this even seems to make the icon work in dev mode in Linux, nice work!

**Comment by jjgalvez** [comment]
I'll make the changes. Do I update the pull request or submit a new one ?




Get Outlook for Android







On Wed, Jul 11, 2018 at 7:25 PM -0400, "Chris Knepper" <notifications@github.com> wrote:












Looks good in Arch as well, though to get it to run I had to do:


sudo pacman -S libxss

sudo pacman -S gconf


And I'm an Arch super-noob, though those dependencies probably also apply to any Electron apps for Arch in dev mode.

**Comment by ajivoin** [comment]
@jjgalvez if you just push your changes it will update ther pull request

**Comment by jjgalvez** [comment]
Cool thanks




Get Outlook for Android







On Wed, Jul 11, 2018 at 9:24 PM -0400, "Andrew Jivoin" <notifications@github.com> wrote:












@jjgalvez if you just push your changes it will update ther pull request

**Comment by jjgalvez** [comment]
I updated app.js and background.js to move the icon creation. BTW if you add pacman to your build stanza in package.json and then install in arch via sudo pacman -U  then the dependencies are managed correctly.

**Comment by chrisknepper** [comment]
Sounds good, I'll get this in a release shortly. Thanks for your work @jjgalvez and @ajivoin !

**Comment by jjgalvez** [comment]
No problem glad to help


Jose




Get Outlook for Android







On Thu, Jul 12, 2018 at 9:34 PM -0400, "Chris Knepper" <notifications@github.com> wrote:












Sounds good, I'll get this in a release shortly. Thanks for your work @jjgalvez and @ajivoin !

**Pull request merged**